packages/backend-app-api/src/wiring/BackendInitializer.ts

```diff
@@ -74,12 +83,18 @@ const instanceRegistry = new (class InstanceRegistry {
 
   #exitHandler = async () => {
     try {
-      const results = await Promise.allSettled(
-        Array.from(this.#instances).map(b => b.stop()),
-      );
-      const errors = results.flatMap(r =>
-        r.status === 'rejected' ? [r.reason] : [],
-      );
+      // This signals to the healthcheck service that the process is shutting down
+      process.exitCode = 0;
+
+      const results = await Promise.race([
+        // Give the backend 30 seconds to shut down, then force exit
+        new Promise(resolve => setTimeout(resolve, 30000)),
+        Promise.allSettled(Array.from(this.#instances).map(b => b.stop())),
```

Comment:
Tbh, I'm kinda wondering if all we need to do is add a wait to the `stop()` after we have called the shutdown hooks. Basically we'd make it a `backend.postShutdownWait` configuration or something like that, just a silly little wait for a certain amount of time after the shutdown hooks have run successfully. The goal isn't really to drain traffic but to make sure that we won't receive any new one, so I think that'd be alright.

Reply:
We decided to implement a new before shutdown lifecycle hook, it will be called before plugin and services are stopped.